docs/new_flux_ideas_and_guide.md

```diff
@@ -0,0 +1,985 @@
+
+```
+@my_data.h2o_temperature{location, state, bottom_degrees, time > -3h}
+    |> selector(fn: max(bottom_degrees), by: ["state"], window: 1h, windowColumn: "window")
+
+// if you include window_column, the result will put the window timestamp for each selected row in that column
+```
+
+Produces this result:
+
+| location | state | bottom_degrees | time | window |
+| --- | --- | --- | --- | --- |
+| santa_monica | CA | 50 | 2020-02-22T15:01 | 2020-02-22T16:00 |
+| coyote_creek | CA | 51 | 2020-02-22T16:05 | 2020-02-22T17:00 |
+| santa_monica| CA | 48 | 2020-02-22T17:01 | 2020-02-22T18:00 |
+| santa_monica| CA | 48 | 2020-02-22T17:31 | 2020-02-22T18:00 |
+| santa_monica| CA | 48 | 2020-02-22T17:46 | 2020-02-22T18:00 |
+| puget_sound | WA | 40 | 2020-02-22T17:10 | 2020-02-22T18:00 |
+| puget_sound | WA | 40 | 2020-02-22T17:40 | 2020-02-22T18:00 |
+
+What this does is for each 1 hour period, it groups the rows by state and then selects the rows with the max value. 
+Because we included the `windowColumn` argument, that is included in the result with the ending timestamp of the 
+window that each record fell into.
+
+Since so many of the records have tied `max(bottom_degrees)` values, we see that many of the original 
+rows were selected in the result. We can also see the `window` that is inserted to specify 
+which block of time the selector was working across.
+
+The following functions can be used in the **Selector Predicate Block**: `first`, `last`, `min`, `max`, `top`, `bottom`
+
+#### Calling Aggregates without the Aggregate function
+The `aggregate` function provides a shorthand for computing multiple aggregates at a time. However, the 
+functions passed in can also be called on tables. For example:
+
+```
+// get the most recently written record
+@my_data.h2o_temperature |> last()
+
+// get the most recently written record for each location
+@my_data.h2o_temperature |> last(by: ["location"])
+
+// get the average for each hour for each location
+@my_data.h2o_temperature{time > -3h} |> mean(by: ["location"], window: 1h)
+``` 
+
+Selectors can only be called with the `select` function.
+
+#### Count, Distinct, and CountDistinct
+The `count` function is an aggregate while the `distinct` function is a selector. The `countDistinct` function is 
+a helper function that will combine operations for you.
+ 
+```
+@my_data.h2o_temperature{state, time > -3h}
+    |> select(fn: distinct(state), window: 1h, windowColumn: "time")
+```
+
+Would return the following result
+
+| state | time
+| --- | --- |
+| CA | 2020-02-22T16:00
+| CA | 2020-02-22T17:00
+| CA | 2020-02-22T18:00
+| WA | 2020-02-22T18:00
+
+We can see that CA reported in all three hours while both CA and WA reported in only the last hour. Also see that 
+distinct discards columns that aren't in the `by` argument or the `distinct` column itself. So we use the `time` 
+name for where we put the window markers for each distinct state. You could then count the number of states reporting 
```

Comment:
It's not clear to me how the first two sentences explain why I would use `time` for the `windowColumn`.